egs/dihard_2018/v2/run.sh

```diff
@@ -0,0 +1,323 @@
+  for threshold in -0.5 -0.4 -0.3 -0.2 -0.1 -0.05 0 0.05 0.1 0.2 0.3 0.4 0.5; do
+    diarization/cluster.sh --cmd "$train_cmd --mem 4G" --nj 20 \
+      --threshold $threshold $nnet_dir/xvectors_dihard_2018_dev/plda_scores \
+      $nnet_dir/xvectors_dihard_2018_dev/plda_scores_t$threshold
+
+    md-eval.pl -1 -c 0.25 -r data/dihard_2018_dev/rttm \
+     -s $nnet_dir/xvectors_dihard_2018_dev/plda_scores_t$threshold/rttm \
+     2> $nnet_dir/tuning/dihard_2018_dev_t${threshold}.log \
+     > $nnet_dir/tuning/dihard_2018_dev_t${threshold}
+
+    der=$(grep -oP 'DIARIZATION\ ERROR\ =\ \K[0-9]+([.][0-9]+)?' \
+      $nnet_dir/tuning/dihard_2018_dev_t${threshold})
+    if [ $(echo $der'<'$best_der | bc -l) -eq 1 ]; then
+      best_der=$der
+      best_threshold=$threshold
+    fi
+  done
+  echo "$best_threshold" > $nnet_dir/tuning/dihard_2018_dev_best
+
+  diarization/cluster.sh --cmd "$train_cmd --mem 4G" --nj 20 \
+    --threshold $(cat $nnet_dir/tuning/dihard_2018_dev_best) \
+    $nnet_dir/xvectors_dihard_2018_dev/plda_scores $nnet_dir/xvectors_dihard_2018_dev/plda_scores
+
+  # Cluster dihard_2018_eval using the best threshold found for xvectors_dihard_2018_dev. 
+  # xvectors_dihard_2018_dev is used as the dataset to tune the parameters. 
+  diarization/cluster.sh --cmd "$train_cmd --mem 4G" --nj 20 \
+    --threshold $(cat $nnet_dir/tuning/dihard_2018_dev_best) \
+    $nnet_dir/xvectors_dihard_2018_eval/plda_scores $nnet_dir/xvectors_dihard_2018_eval/plda_scores
+
+  mkdir -p $nnet_dir/results
+  # Compute the DER on the dihard_2018_eval. 250ms of a speaker transition is tolerated
+  # and overlapping segments are ignored. 
+  md-eval.pl -1 -c 0.25 -r data/dihard_2018_eval/rttm \
+    -s $nnet_dir/xvectors_dihard_2018_eval/plda_scores/rttm 2> $nnet_dir/results/threshold.log \
+    > $nnet_dir/results/DER_threshold.txt
+  der=$(grep -oP 'DIARIZATION\ ERROR\ =\ \K[0-9]+([.][0-9]+)?' \
+    $nnet_dir/results/DER_threshold.txt)
+  echo "Using supervised calibration, DER: $der%"
+fi
+
+# Cluster the PLDA scores using the oracle number of speakers
+if [ $stage -le 13 ]; then
+  # In this section, we show how to do the clustering if the number of speakers
+  # (and therefore, the number of clusters) per recording is known in advance.
+  diarization/cluster.sh --cmd "$train_cmd --mem 4G" \
+    --reco2num-spk data/dihard_2018_eval/reco2num_spk \
+    $nnet_dir/xvectors_dihard_2018_eval/plda_scores $nnet_dir/xvectors_dihard_2018_eval/plda_scores_num_spk
+
+  mkdir -p $nnet_dir/results
+  md-eval.pl -1 -c 0.25 -r data/dihard_2018_eval/rttm \
+    -s $nnet_dir/xvectors_dihard_2018_eval/plda_scores_num_spk/rttm 2> $nnet_dir/results/num_spk.log \
+    > $nnet_dir/results/DER_num_spk.txt
+  der=$(grep -oP 'DIARIZATION\ ERROR\ =\ \K[0-9]+([.][0-9]+)?' \
+    $nnet_dir/results/DER_num_spk.txt)
+  echo "Using the oracle number of speakers, DER: $der%"
+fi
+
+# Score the overlapping part and do not tolerate errors within 250ms of a speaker transition
```

Comment:
Could you clarify that this is how the RTTMs were scored during the DIHARD challenge? This would be helpful info for anyone who wants to compare results.